File: examples/geoquery.py
#!/usr/bin/env python3

# Disable "Invalid constant name"
# pylint: disable=C0103

import pandas as pd
import ukcensusapi.Nomisweb as Api

API = Api.Nomisweb("./")

print("Nomisweb census data geographical query example")
print("See README.md for details on how to use this package")

# In the previous example we had a predefined query using Leeds at MSOA resolution,
# but we want to expand the geographical area and refine the resolution
TABLE = "NM_618_1"
query_params = {}
query_params["CELL"] = "7...13"
query_params["date"] = "latest"
query_params["RURAL_URBAN"] = "0"
query_params["select"] = "GEOGRAPHY_CODE,CELL,OBS_VALUE"
query_params["geography"] = "1245710558...1245710660,1245714998...1245714998,1245715007...1245715007,1245715021...1245715022"
query_params["MEASURES"] = "20100"

# Define the new coverage area in terms of local authorities
coverage = ["Leeds", "Bradford"]
# Define the new resolution
resolution = Api.Nomisweb.OA
# Convert the coverage area into nomis codes
coverage_codes = API.get_lad_codes(coverage)
# replace the geography value in the query
query_params["geography"] = API.get_geo_codes(coverage_codes, resolution)
# get the data
KS401FINE = API.get_data(TABLE, query_params)
print(KS401FINE.head(5))

# Now widen the coverage to England & Wales and coarsen the resolution to LA
coverage_codes = [Api.Nomisweb.EnglandWales]
resolution = Api.Nomisweb.LAD
query_params["geography"] = API.get_geo_codes(coverage_codes, resolution)
# get the data
KS401BROAD = API.get_data(TABLE, query_params)
print(KS401BROAD.head(5))

File: examples/geoquery.R

# this example code is an interim solution pending proper R-python integration
library("UKCensusAPI")

cacheDir = "./"
queryUrl = "https://www.nomisweb.co.uk/api/v01/dataset/NM_618_1.data.tsv?CELL=7...13&MEASURES=20100&RURAL_URBAN=0&date=latest&geography=1245710558...1245710660%2C1245714998...1245714998%2C1245715007...1245715007%2C1245715021...1245715022&select=GEOGRAPHY_CODE%2CCELL%2COBS_VALUE"

# Define the new region and resolution
coverage = c("Leeds","Bradford")
resolution = 299 # OA - see NomiswebApi.py

# Modify the url
newQueryUrl = modifyGeography(queryUrl, coverage, resolution)

# Fetch the new data
KS401EW = getData(newQueryUrl, cacheDir)

